src/python/zquantum/core/interfaces/ansatz.py

```diff
@@ -87,6 +87,9 @@ def _generate_circuit(self, params: Optional[np.ndarray] = None) -> Circuit:
         Will return parametrized circuits if no parameters are passed and the ansatz
         supports parametrized circuits.
 
+        Ansatzes that implement this method must not ignore parameters if parameters
```

Comment:
I think this is a pretty logical thing, so I'm not really sure if it's worth mentioning here.

Reply:
Some ansatzes don't do it though 🤔

Reply:
Then leave it be 🤷